libs/portals/admin/air-discount-scheme/src/screens/Overview/Overview.loader.ts

```diff
@@ -18,16 +19,18 @@ import {
 
 const TODAY = new Date()
 
-export const transformDate = (val: string) => {
+export const transformDate = (val: string, start: boolean) => {
   if (isValidDate(new Date(val)) && val.indexOf('.') === 2) {
-    return parseDate(val, 'dd.MM.yyyy', endOfDay(new Date()))
+    const value = parseDate(val, 'dd.MM.yyyy', new Date())
+
+    return start ? startOfDay(value) : endOfDay(value)
   }
 
   if (isValidDate(new Date(val))) {
     return val
   }
-
-  return parseDate(val, 'dd.MM.yyyy', endOfDay(new Date()))
+  const value = parseDate(val, 'dd.MM.yyyy', new Date())
+  return start ? startOfDay(value) : endOfDay(value)
```

Comment:
This is exactly the same is in the first if statement. And why is L30 only returning val which is string when other paths return either startOf or endOf day which is a date object.
What is the goal of this function and I feel like it could be refactored to be more simple and readable.

Reply:
Was originally supposed to handle if someone was setting a wrong date manually into query string, but lets move from that, now it just returns the string back if the date is not valid.